src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
+import java.util.HashMap;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTORY = 0.75f;
+    private static final float LOAD_FACTORY = DEFAULT_LOAD_FACTORY;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int INCREASE_TABLE = 2;
+    private static final int HASH_SHIFT = 16;
+    private int initialCapacity = DEFAULT_CAPACITY;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int size;
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[initialCapacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        threshold = (int) (initialCapacity * LOAD_FACTORY);
+        if (size >= threshold) {
+            resize();
+        }
+        int index = Math.abs(hash(key)) % table.length;
+        var current = table[index];
+        if (table[index] == null) {
+            table[index] = new Node<>(index, key, value, null);
```

Comment:
Using Node constructor with a hash parameter can be confusing. The hash of a node is typically calculated internally and is not meant to be set externally. Consider removing the hash parameter from the constructor and calculating the hash inside the Node class.